src/sinks/loki/sink.rs

```diff
@@ -311,21 +312,41 @@ impl RecordFilter {
     }
 }
 
-#[derive(Clone)]
 pub struct LokiSink {
     acker: Acker,
     request_builder: LokiRequestBuilder,
     pub(super) encoder: EventEncoder,
     batch_settings: BatcherSettings,
     out_of_order_action: OutOfOrderAction,
-    service: LokiService,
+    service: tower::util::BoxService<LokiRequest, LokiResponse, crate::Error>,
 }
 
 impl LokiSink {
     #[allow(clippy::missing_const_for_fn)] // const cannot run destructor
     pub fn new(config: LokiConfig, client: HttpClient, cx: SinkContext) -> crate::Result<Self> {
         let compression = config.compression;
 
+        // if Vector is configured to allow events with out of order timestamps, then then we can
+        // safely enable concurrency settings.
+        //
+        // For rewritten timestamps, we use a static concurrency of 1 to avoid out-of-order
+        // timestamps across requests. We used to support concurrency across partitions (Loki
+        // streams) but this was lost in #9506. Rather than try to re-add it, since Loki no longer
+        // requires in-order processing for version >= 2.4, instead we just keep the static limit
+        // of 1 for now.
+        let request_limits = match config.out_of_order_action {
+            OutOfOrderAction::Accept => config.request.unwrap_with(&Default::default()),
+            OutOfOrderAction::Drop | OutOfOrderAction::RewriteTimestamp => {
```

Comment:
The PR comment says it changes the behavior for `Accept` and `Drop`, but here they are separate cases. Is that right?

Reply:
Ah, yes, I originally thought `Drop` should be able to be concurrent too, but realized it cannot but neglected to update the comment.